app/src/main/java/com/sapuseven/untis/activities/MainActivity.kt

```diff
@@ -575,12 +575,16 @@ class MainActivity :
 
 			lines.add(startTimeInt)
 			lines.add(endTimeInt)
+			labels.add(hour.label)
 		}
 
 		if (!PreferenceUtils.getPrefBool(preferences, "preference_timetable_range_index_reset"))
 			weekView.hourIndexOffset = (range?.first ?: 1) - 1
 		weekView.hourLines = lines.toIntArray()
-
+		weekView.hourLabels = labels.toTypedArray().let { hourLabelArray ->
+			if(hourLabelArray.joinToString("")=="") IntArray(labels.size, fun(idx:Int):Int{return idx+1}).map{it.toString()}.toTypedArray()
+			else hourLabelArray
+		}
```

Comment:
(see above)
```suggestion
		weekView.hourLabels = labels.toTypedArray()
```

Reply:
I thought that, if there's a label missing for one hour, f.e. "0" does not have a label, but every other hour does, it would show "1, 1, 2, 3, ..." instead of just not putting a label for the first one. Schools do some weird stuff, so it wouldn't surprise me if that was a thing.

Reply:
I see... yeah you're right.